code-challenges.js

```diff
@@ -39,3 +115,20 @@ var numbersToAdd2 = [0, 7, -8, 12]
 // Expected output: [0, 7, -1, 11]
 var numbersToAdd3 = []
 // Expected output: []
+
+
+
+const sum = (array) => {
+  let newarr= array.map((value, index) => array.slice(0,index + 1).reduce((a, b) => a + b))
+return newarr
+}
+
+console.log("Excpected output 5.1 : [2, 6, 51, 60]")
+console.log("Challenge 5.1:",sum(numbersToAdd1))
+
+console.log("Excpected output 5.2 : [0, 7, -1, 11]")
+console.log("Challenge 5.2:",sum(numbersToAdd2))
+
+console.log("Expected output 5.3 : []")
+console.log("Challenge 5.3:",sum(numbersToAdd3))
+
```

Comment:
Love it. Just a small comment, any time you are creating a variable just to return it, you can always skip that step and just return the code itself. What you are doing is perfectly correct, just a thought as you move forward in your coding journey.